Test failure if SATPY_CONFIG_PATH set

**Describe the bug**

If `SATPY_CONFIG_PATH` is set while running `pytest`, `TestPluginsConfigs::test_get_plugin_configs` fails with `AssertionError`.

**To Reproduce**

```bash
SATPY_CONFIG_PATH= pytest -k test_get_plugin_configs
```

**Expected behavior**

I expect tests to pass regardless of environmental variables.

**Actual results**

```
===================================================== test session starts =====================================================
platform linux -- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/gholl/checkouts/satpy
plugins: subprocess-1.0.0, socket-0.3.5, cov-2.10.1
collected 1110 items / 1109 deselected / 1 selected

test_config.py F                                                                                                        [100%]

========================================================== FAILURES ===========================================================
_________________________________________ TestPluginsConfigs.test_get_plugin_configs __________________________________________

self = <satpy.tests.test_config.TestPluginsConfigs testMethod=test_get_plugin_configs>
iter_entry_points = <MagicMock name='iter_entry_points' id='140066037209456'>

    @mock.patch('satpy._config.pkg_resources.iter_entry_points')
    def test_get_plugin_configs(self, iter_entry_points):
        """Check that the plugin configs are looked for."""
        import pkg_resources
        ep = pkg_resources.EntryPoint.parse('example_composites = satpy_cpe')
        ep.dist = pkg_resources.Distribution.from_filename('satpy_cpe-0.0.0-py3.8.egg')
        ep.dist.module_path = os.path.join(os.path.sep + 'bla', 'bla')
        iter_entry_points.return_value = [ep]

        from satpy._config import get_entry_points_config_dirs
        dirs = get_entry_points_config_dirs('satpy.composites')
>       self.assertListEqual(dirs, [os.path.join(ep.dist.module_path, 'satpy_cpe', 'etc')])
E       AssertionError: Lists differ: ['/bla/bla/satpy_cpe/etc', ''] != ['/bla/bla/satpy_cpe/etc']
E
E       First list contains 1 additional elements.
E       First extra element 1:
E       ''
E
E       - ['/bla/bla/satpy_cpe/etc', '']
E       ?                          ----
E
E       + ['/bla/bla/satpy_cpe/etc']

test_config.py:107: AssertionError
====================================================== warnings summary =======================================================
../../../../../../data/gholl/miniconda3/envs/py38b/lib/python3.8/site-packages/pyproj/crs/crs.py:543
  /data/gholl/miniconda3/envs/py38b/lib/python3.8/site-packages/pyproj/crs/crs.py:543: UserWarning: You will likely lose important projection information when converting to a PROJ string from another format. See: https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems
    proj_string = self.to_proj4()

reader_tests/test_mviri_l1b_fiduceo_nc.py:531
  /home/gholl/checkouts/satpy/satpy/tests/reader_tests/test_mviri_l1b_fiduceo_nc.py:531: PytestUnknownMarkWarning: Unknown pytest.mark.file_handler_data - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.file_handler_data(mask_bad_quality=False)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=================================================== short test summary info ===================================================
FAILED test_config.py::TestPluginsConfigs::test_get_plugin_configs - AssertionError: Lists differ: ['/bla/bla/satpy_cpe/etc'...
```

**Environment Info:**
 - OS: openSUSE 15.0
 - Satpy Version: v0.25.1-46-g653afc08

**Additional context**

The test probably needs a pytest `monkeypatch` fixture to delete `SATPY_CONFIG_PATH` from the environment (if present).

--- a/satpy/tests/test_config.py
+++ b/satpy/tests/test_config.py
@@ -102,9 +102,12 @@ class TestPluginsConfigs(unittest.TestCase):
         ep.dist.module_path = os.path.join(os.path.sep + 'bla', 'bla')
         iter_entry_points.return_value = [ep]
 
+        import satpy
         from satpy._config import get_entry_points_config_dirs
-        dirs = get_entry_points_config_dirs('satpy.composites')
-        self.assertListEqual(dirs, [os.path.join(ep.dist.module_path, 'satpy_cpe', 'etc')])
+        # don't let user env vars effect results
+        with satpy.config.set(config_path=[]):
+            dirs = get_entry_points_config_dirs('satpy.composites')
+            self.assertListEqual(dirs, [os.path.join(ep.dist.module_path, 'satpy_cpe', 'etc')])
 
 
 class TestConfigObject:

--- a/satpy/tests/test_config.py
+++ b/satpy/tests/test_config.py
@@ -104,7 +104,7 @@ class TestPluginsConfigs(unittest.TestCase):
 
         import satpy
         from satpy._config import get_entry_points_config_dirs
-        # don't let user env vars effect results
+        # don't let user env vars affect results
         with satpy.config.set(config_path=[]):
             dirs = get_entry_points_config_dirs('satpy.composites')
             self.assertListEqual(dirs, [os.path.join(ep.dist.module_path, 'satpy_cpe', 'etc')])

